src/commands/resetIndex.js

```diff
@@ -0,0 +1,65 @@
+import path from 'path'
+
+import { GitIndexManager } from '../managers/GitIndexManager.js'
+import { GitObjectManager } from '../managers/GitObjectManager.js'
+import { FileSystem } from '../models/FileSystem.js'
+
+import { readObject } from './readObject'
+import { log } from './log'
+
+/**
+ * Reset a file from the git index (aka staging area)
+ *
+ * @link https://isomorphic-git.github.io/docs/resetIndex.html
+ */
+export async function resetIndex ({
+  dir,
+  gitdir = path.join(dir, '.git'),
+  fs: _fs,
+  filepath,
+  ref = 'HEAD'
+}) {
+  try {
+    const fs = new FileSystem(_fs)
+    const tree = await log({ gitdir, fs, ref, depth: 1 })
+    let oid = tree[0] && tree[0].tree
+    let workdirOid
+    if (oid) {
```

Comment:
I suggest instead, `if (!oid) throw new GitError(E.ResolveRefError, {ref})`

Reply:
`oid` can be undefined when `ref` not exists or repository is empty

Reply:
I say if `ref` is not accessible it should throw this error. 😁

Reply:
When the repository is empty `ref` will be accessible?

Reply:
Besides it depends to the `log` command more, but `theGitRefManager.resolve` can return only an exception